app/src/main/ui/src/hooks/useGroups.tsx

```diff
@@ -0,0 +1,8 @@
+import {GroupsModel} from "../services/Groups";
+import {QueryOptions, StateSetter, useAPI} from "./useAPI";
+import {GroupsService} from "../services/Groups";
+
+export const useGroups = (opts: QueryOptions): [GroupsModel | undefined, boolean, StateSetter<GroupsModel>] => {
```

Comment:
Here you use `GroupsModel | undefined` and on useUsers you only use `UsersModel`. Is this intentional? What are the reasons to allow or not allow undefined?